From: =?utf-8?q?Rog=C3=A9rio_Brito?= <rbrito@ime.usp.br>
Date: Sat, 21 Feb 2015 18:28:30 -0200
Subject: Makefile: Don't bomb compilation on 32/64 bits.
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Signed-off-by: Rogério Brito <rbrito@ime.usp.br>
---
 makefile | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/makefile b/makefile
index a8933b7..6b1fa0e 100644
--- a/makefile
+++ b/makefile
@@ -45,16 +45,7 @@ ALLREQS=ptr.cmx ptr-c$O crc.cmx list2.cmx expandarray.cmx c_part$O unicode.cmx u
 
 MP3PACKER=$(ALLREQS) mp3packer.cmx
 
-# The -ffast-math line is only to prevent this from running on a 64-bit system. Use "make mp3packer64" instead
-# That line has no bearing on the actual program; it may be removed if it causes problems
 mp3packer: depend $(MP3PACKER)
-	$(ocaml) -ffast-math
-	$(ocaml) -o $@$(EXE) $(MP3PACKER)
-
-# The fail_on_32bit line is only to prevent this from running on a 32-bit system. Use "make mp3packer" or just "make" instead
-# The source is a .txt file since otherwise ocamldep would try to parse the file on the 32-bit build and fail
-mp3packer64: depend $(MP3PACKER)
-	$(ocaml) -c -impl fail_on_32bit.txt
 	$(ocaml) -o $@$(EXE) $(MP3PACKER)
 
 
