Description: Adapt the package to Unix-like systems
Author: Rogério Brito <rbrito@ime.usp.br>
Forwarded: no
Last-Update: 2011-08-19

--- a/makefile
+++ b/makefile
@@ -1,10 +1,11 @@
 # Change this if compiling on Unixy OS
-OBJ_EXT=.obj
-EXE_EXT=.exe
+OBJ_EXT=.o
+EXE_EXT=
 
 ocaml=ocamlopt
 
-all: mp3packer mp3reader
+all: mp3packer
+# mp3reader
 
 mp3packer: crc.cmx list2.cmx expandarray.cmx c_part$(OBJ_EXT) mp3types.cmx pack.cmx mp3read.cmx mp3write.cmx mp3info.cmx mp3framehuffman.cmx mp3frameutils.cmx mp3queue.cmx multiproc.cmx mp3packer.ml
 	ocamlopt -o mp3packer$(EXE_EXT) unix.cmxa str.cmxa crc.cmx list2.cmx expandarray.cmx c_part$(OBJ_EXT) mp3types.cmx pack.cmx mp3read.cmx mp3write.cmx mp3info.cmx mp3framehuffman.cmx mp3frameutils.cmx mp3queue.cmx multiproc.cmx mp3packer.ml
@@ -107,8 +108,8 @@
 
 
 clean:
-	rm -rf ./mp3packer.exe
-	rm -rf ./*.obj
+	rm -rf ./mp3packer
+	rm -rf ./*.o
 	rm -rf ./*.cmi
 	rm -rf ./*.cmx
 	rm -rf ./*.cmo
